lib/sda/processor.go

```diff
@@ -0,0 +1,125 @@
+package sda
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+)
+
+// ServiceProcessor allows for an easy integration of external messages
+// into the Services. You have to embed it into your Service-structer,
+// then it will offer an 'AddMessage'-method that takes a message of type
+// 	func ReceiveMsg(e *network.Entity, msg *anyMessageType)(error, *replyMsg)
+// where 'ReceiveMsg' is any name and 'anyMessageType' will be registered
+// with the network. Once 'anyMessageType' is received by the service,
+// the function 'ReceiveMsg' should return an error and any 'replyMsg' it
+// wants to send.
+type ServiceProcessor struct {
+	functions map[network.MessageTypeID]interface{}
+	Context
+}
+
+// NewServiceProcessor initializes your ServiceProcessor.
+func NewServiceProcessor(c Context) *ServiceProcessor {
+	return &ServiceProcessor{
+		functions: make(map[network.MessageTypeID]interface{}),
+		Context:   c,
+	}
+}
+
+// AddMessage puts a new message in the message-handler
+func (p *ServiceProcessor) AddMessage(f interface{}) error {
```

Comment:
The name `AddMessage` is confusing for me in this context. What do you think of `processor.RegisterHandler(fn)` (so it's clear we expect a function that handles a certain type of message), or `processor.Handler(fn)` (short minimalistic golang-alike version ... )